protocol-impl/src/main/java/io/camunda/zeebe/protocol/impl/record/RecordMetadata.java

```diff
@@ -265,4 +265,56 @@ public String toString() {
         + brokerVersion
         + '}';
   }
+
+  @Override
+  public boolean equals(final Object o) {
```

Comment:
:thought_balloon: I'm not sure about this. It's useful for tests to see if read metadata matches written metadata but I'm wondering if this is dangerous in other places.

Reply:
:question: Do you mean if it'll break some compatibility? I guess it would if we were using it in maps/sets and so on in production, but I don't think so :thinking: 

:wrench: I think you used a generator different than the typical one we used, where we rely on `Objects.hash` and `Objects.equals`. I think IntelliJ calls it the Java 7+ one or something.

Anyway, if our tests pass, then I think this is a non-issue :+1: